### PR TITLE
Everything we call by name needs to be `Pkg.add`ed

### PR DESCRIPTION
**Pull request by ararslan**
This includes Compat, BenchmarkTools, and JSON. Otherwise Julia complains that it can't find those packages.

### CODE REVIEW

**Comment by vchuravy** [comment]
Have a `Project.toml` that you initialise?

**Comment by ararslan** [comment]
Seems like a good idea. @KristofferC, thoughts?

**Comment by KristofferC** [comment]
Project + Manifest that are checked in to this repo and are copied to workers + then instantiated seems like a good idea.

**Comment by ararslan** [comment]
Help generating those would be great. I'm not sure about directly instantiating a manifest though, because we want to have the latest versions of things rather than what was installed at the time the manifest file was created.

**Comment by KristofferC** [comment]
You could track master and Pkg.update I guess.

**Comment by vchuravy** [comment]
Do we really want to have the master version? I would think that we probably want to have a consistent set of dependencies that we update once in a while and not everytime.

**Comment by KristofferC** [comment]
With pre 1.0 spree it probably makes sense. In a post 1.0 world, keeping things as they are might be preferable.

**Comment by ararslan** [comment]
I meant latest tags of the packages, not _latest_ latest, like master. The only things in Nanosoldier land that aren't on tags are Nanosoldier and BaseBenchmarks, since neither are registered and using the bleeding edge is what you want for those.

File: src/jobs/BenchmarkJob.jl [hunk 1]
```diff
@@ -278,6 +278,8 @@ function execute_benchmarks!(job::BenchmarkJob, whichbuild::Symbol)
                 else
                     Pkg.clone(url)
                 end
+                # These are referenced by name so they need to be added explicitly
+                foreach(Pkg.add, ("Compat", "BenchmarkTools", "JSON"))
```
Comment (KristofferC) [inline src/jobs/BenchmarkJob.jl]: `Pkg.add(["Compat", "BenchmarkTools", "JSON"])` is faster :P

Comment (ararslan) [reply src/jobs/BenchmarkJob.jl]: True, but won't work on 0.6.

Comment (KristofferC) [reply src/jobs/BenchmarkJob.jl]: Ah, good point

**Pull request merged**